src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -0,0 +1,424 @@
+                .baseUri(voucherManagementConfig.voucherManagementContactPoint)
+                .body(createVoucherBody)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(expectedStatus).build())
+                .when()
+                .post(voucherManagementConfig.createVoucherEndpoint)
+                .andReturn().asString();
+
+
+        logger.info("Identity Mapper Response: {}", BaseStepDef.response);
+        try {
+            Thread.sleep(10000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public static String generateUniqueNumber(int length) {
+        Random rand = new Random();
+        long timestamp = System.currentTimeMillis();
+        long randomLong = rand.nextLong(100000000);
+        String uniqueNumber = timestamp + "" + randomLong;
+        return uniqueNumber.substring(0, length);
+    }
+
+    @When("I can create an VoucherRequestDTO for voucher activation")
+    public void iCanCreateAnVoucherRequestDTOForVoucherActivation() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("{\n");
+        sb.append("    \"requestID\": \"849324499155\",\n");
+        sb.append("    \"batchID\": \"045155518258\",\n");
+        sb.append("    \"voucherInstructions\": [\n");
+        sb.append("        {\n");
+        sb.append("            \"serialNumber\": \"").append(serialNumber).append("\",\n");
+        sb.append("            \"status\": \"02\"\n");
+        sb.append("        }\n");
+        sb.append("    ]\n");
+        sb.append("}");
+
+        activateVoucherBody = sb.toString();
+    }
+
+    @When("I call the activate voucher API with expected status of {int} and stub {string}")
+    public void iCallTheActivateVoucherAPIWithExpectedStatusOfAndStub(int expectedStatus, String stub) {
+        BaseStepDef.response = RestAssured.given()
+                .header("Content-Type", "application/json")
+                .header("X-CallbackURL", mockServer.getBaseUri()+stub)
+                .header("registeringInstitutionId", registeringInstitutionId)
+                .queryParam("command", "activate")
+                .baseUri(voucherManagementConfig.voucherManagementContactPoint)
+                .body(activateVoucherBody)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(expectedStatus).build())
+                .when()
+                .put(voucherManagementConfig.voucherLifecycleEndpoint)
+                .andReturn().asString();
+
+
+        logger.info("Identity Mapper Response: {}", BaseStepDef.response);
+    }
+
+
+    @Then("I should be able to verify that the {string} method to {string} endpoint received a request with required parameter in create voucher callback body")
+    public void iShouldBeAbleToVerifyThatTheMethodToEndpointReceivedARequestWithRequiredParameterInCreateVocuherCallbackBody(String method, String endpoint) {
```

Comment:
method names are too long, can these be shorter?

Reply:
these methods are auto generated when we create new step def